spec/controllers/works/drafts_spec.rb

```diff
@@ -83,30 +71,25 @@
       put :post_draft, id: random_work.id
       # There is code to return a different message in the action, but it is unreachable using a web request
       # as the application_controller redirects the user first
-      expect(flash[:error]).to start_with "Sorry, you don't have permission to access"
+      it_redirects_to_with_error(work_path(random_work),\
+                                 "Sorry, you don't have permission to access the page you were trying to reach.")
     end
 
     context "if the work is already posted" do
       it "should display an error" do
         drafts_user_work = create(:work, authors: [drafts_user.default_pseud], posted: true)
         put :post_draft, id: drafts_user_work.id
-        expect(flash[:error]).to eq "That work is already posted. Do you want to edit it instead?"
-      end
-
-      it "should redirect to the user work edit page" do
-        drafts_user_work = create(:work, authors: [drafts_user.default_pseud], posted: true)
-        put :post_draft, id: drafts_user_work.id
-        expect(response).to have_http_status(:redirect)
-        expect(response).to redirect_to edit_user_work_path(drafts_user, drafts_user_work)
+        it_redirects_to_with_error(edit_user_work_path(drafts_user, drafts_user_work),\
```

Comment:
Shouldn't need a \, and the description of the example should be updated now that you've combined two examples